`init` should not move certain scripts #97

Miklet opened the pull request:
<!--
Thanks for your interest in the project. Bugs filed and PRs submitted are appreciated!

Please make sure that you are familiar with and follow the Code of Conduct for
this project (found in the CODE_OF_CONDUCT.md file).

Also, please make sure you're familiar with and follow the instructions in the
contributing guidelines (found in the CONTRIBUTING.md file).

If you're new to contributing to open source projects, you might find this free
video course helpful: http://kcd.im/pull-request

Please fill out the information below to expedite the review and (hopefully)
merge of your pull request!
-->

<!-- What changes are being made? (What feature/bug is being fixed here?) -->
**What**: Leave npm lifecycle scripts and other core scripts untouched, convert `npm run` commansd to `nps`.

<!-- Why are these changes necessary? -->
**Why**: To make migrating from npm to nps easier

<!-- How were these changes implemented? -->
**How**: 
* Added method which selects core scripts from package.json
* Moved stringify object method to separate module
* Added converting `npm run` scripts to `nps`

<!-- Have you done all of these things?  -->
**Checklist**:
<!-- add "N/A" to the end of each line that's irrelevant to your changes -->
<!-- to check an item, place an "x" in the box like so: "- [x] Documentation" -->
- [ ] Documentation N/A
- [x] Tests 
- [x] Ready to be merged <!-- In your opinion, is this ready to be merged as soon as it's reviewed? -->
- [ ] Added myself to contributors table <!-- this is optional, see the contributing guidelines for instructions --> N/A

Hi @kentcdodds . My work is still in progress, tests will be added soon, but first want to make sure that my changes are suitable and also becasue at this point it is easier to revert some changes 😄

kentcdodds commented on src/bin-utils/initialize/stringify-object.js:
```diff
@@ -0,0 +1,25 @@
+import {isPlainObject} from 'lodash'
+
+function stringifyObject(object, indent) {
```
Could we add a comment on this? I'm starting to forget what all this code is supposed to do 🙃

Miklet commented:
Alright, I have rebased everything into one commit, added and updated tests, also added a comment as you wanted 😃 Now it is ready to merge in my opinion.

Miklet commented on src/bin-utils/initialize/__tests__/index.js:
```diff
@@ -131,6 +131,33 @@ test(`initialize without any scripts should successfully create an empty package
   })
 })
 
+test('initialize without any core scripts and should not add any core scripts to package.json', () => {
```
Ahhh wrong test name, gonna fix it fast

Miklet replied:
Done!
